flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/utils/TestingJobMasterGateway.java

```diff
@@ -436,10 +438,18 @@ public void acknowledgeCheckpoint(
             ExecutionAttemptID executionAttemptID,
             long checkpointId,
             CheckpointMetrics checkpointMetrics,
-            TaskStateSnapshot subtaskState) {
+            SerializedValue<TaskStateSnapshot> subtaskState) {
+        TaskStateSnapshot snapshot;
+        try {
+            snapshot =
+                    subtaskState == null
+                            ? null
+                            : subtaskState.deserializeValue(ClassLoader.getSystemClassLoader());
```

Comment:
We could add some utility methods to the TaskStateSnapshot for the serialization so this becomes less  noisy.

Reply:
Good point.